elmslie-core/src/main/java/vivid/money/elmslie/core/store/MappingActor.kt

```diff
@@ -1,21 +1,21 @@
 package vivid.money.elmslie.core.store
 
+import kotlinx.coroutines.CancellationException
 import vivid.money.elmslie.core.config.ElmslieConfig
 
-/**
- * Contains internal event mapping utilities
- */
+/** Contains internal event mapping utilities */
 interface MappingActor<Event> {
 
     companion object {
         private val logger = ElmslieConfig.logger
     }
 
-    fun Throwable.logErrorEvent(
-        errorMapper: (Throwable) -> Event?
-    ): Event? = errorMapper(this).also {
-        logger.nonfatal(error = this)
-        logger.debug("Failed app state: $it")
+    fun Throwable.logErrorEvent(errorMapper: (Throwable) -> Event?): Event? {
+        val error = (this as? CancellationException)?.cause ?: this
```

Comment:
if we have CancellationException it's an error. CancellationException should never be caught by us